### ISSUE TITLE
train parameters of lr-model.ckpt

### ISSUE THREAD

**Issue by showger** [bhimrazy/diabetic-retinopathy-detection#1]
May I ask what parameters were used to train the lr-model.ckpt? Why do the prediction results from the checkpoint I trained differ from those on the website https://bhimrazy-diabetic-retinopathy-detection.hf.space/?

**Comment by bhimrazy** [comment]
> May I ask what parameters were used to train the lr-model.ckpt? Why do the prediction results from the checkpoint I trained differ from those on the website https://bhimrazy-diabetic-retinopathy-detection.hf.space/?

Hi @showger, great question!

The difference in prediction results likely comes down to the training settings and hyperparameters used. Even small variations—like different data splits, learning rates, the number of epochs, seed etc—can lead to different model checkpoints and performance.

The version I deployed on [Hugging Face Spaces](https://bhimrazy-diabetic-retinopathy-detection.hf.space/) wasn’t that optimized. I think, I trained it for just around 20 epochs and didn’t apply much optimizations too.

I recommend preparing an evaluation set that was not used during training and evaluating your trained model on it. This will give a clearer picture of how well your model is performing, rather than comparing it directly with the results from my deployed version.

Let me know if this helps, and feel free to raise any other questions you might have.
Thanks! 😊

**Comment by showger** [comment]
@bhimrazy Thank you for your response. I'll proceed to fine-tune certain parameters and conduct several more training runs - this iterative process is also an integral part of my learning journey with the model development.

**Comment by bhimrazy** [comment]
That sounds like a great plan, @showger! 😊
Feel free to reach out if you run into anything or want feedback on results. Happy learning! 🚀
